src/complex/DataStructure/DataStructure.cpp

```diff
@@ -187,6 +196,11 @@ Result<LinkedPath> DataStructure::makePath(const DataPath& path)
     return {LinkedPath(this, pathIds)};
   } catch(const std::exception& e)
   {
+    for(const auto& id : createdIds)
+    {
+      removeData(id);
+    }
+
     return complex::MakeErrorResult<LinkedPath>(-2, "Exception thrown when attempting to create a path in the DataStructure");
```

Comment:
Please also incorporate that actual value of 'path' into the error message. If there are ways to better determine the type of exception thrown that would also help in the error message.